pkg/tarmak/provider/amazon/amazon.go

```diff
@@ -576,16 +582,31 @@ func (a *Amazon) verifyInstanceTypes() error {
 		return err
 	}
 
+	constBackoff := backoff.NewConstantBackOff(time.Second * 5)
+	var retries uint64 = 5
+	b := backoff.WithMaxTries(constBackoff, retries)
+
 	for _, instance := range a.tarmak.Cluster().InstancePools() {
 		instanceType, err := a.InstanceType(instance.Config().Size)
 		if err != nil {
 			return err
 		}
 
-		if err := a.verifyInstanceType(instanceType, instance.Zones(), svc); err != nil {
-			result = multierror.Append(result, err)
+		verifyInstanceTypeFunc := func() error {
+			err = a.verifyInstanceType(instanceType, instance.Zones(), svc)
+			if err != nil {
+				// return an error if we're exceeding AWS API request limits
+				if strings.Contains(err.Error(), "RequestLimitExceeded") {
+					return err
+				}
+			}
+			return nil
```

Comment:
This is not quite the functionality that we want.

Backoff will continue to try the function so long as the function returns an error and the context - in this case a max try - is good. In it's current state, an error that is not a rate limit will return fine when this is not the case. Instead here we should use a context where an error that is not a rate limit should return with that error immediately.

```
ctx, cancelRetries := context.WithCancel(context.Background()) 
b := backoff.WithContext(expBackoff, ctx)
```